definitions/npm/ramda_v0.26.x/flow_v0.104.x-/test_ramda_v0.26.x_list.js

```diff
@@ -822,4 +818,71 @@ const str: string = "hello world";
     s: a,
     y: b
   }))([1, 2, 3])(["1", "2", "3"]);
+
+  describe('nth', () => {
+    it('should get a maybe type back on input is an array of given types', () => {
+      const nthxs1: ?number = nth(2, [1,2,3]);
+      const nthxs2: ?number = nth(2)([1,2]);
+      const nthxs3: ?number = nth(3)([1,2]);
+      const nthxs4: ?string = nth(2, ["curry"]);
+      const nthxs5: ?string = nth(2)(["curry", "bean"]);
+      const nthxs6: ?string = nth(3)(['foo', 'bar']);
+    });
+
+    it('should get a string back when input is string', () => {
+      const nthxs: string = nth(2)("curry");
+      const nthxs1: string = nth(2, "curry");
+    });
+
+    it('should take read only array as second argument', () => {
+      const readOnlyArray: $ReadOnlyArray<number>  = [1, 2, 3];
+      const nthxs = nth(2)(readOnlyArray)
+      const nthxs1 = nth(2, readOnlyArray)
+    });
+
+    it('should take a mutable array as second arugment', () => {
+      const array: Array<string> = ['foo', 'bar'];
+      const nthxs = nth(1)(array)
+      const nthxs1 = nth(1, array)
+
+      const tutple:[number, boolean] = [1, true]
+      const nthxs2 = nth(1)(array)
+    });
+
+    it('should fail on expecting return type to be a maybe type of the element of the input list', () => {
+      const arr: Array<number> = [1, 2, 3]
+      //$ExpectError
+      const nthxs: ?string = nth(2)(arr);
+      //$ExpectError
+      const nthxs1: ?boolean = nth(1)(arr);
+    });
+
+    it('should fail on expecting return type to mismatch the input array type', () => {
+      const arr: Array<number> = [1, 2, 3]
+      //$ExpectError
+      const nthxs: string = nth(2, arr);
+      //$ExpectError
+      const nthxs1: string = nth(2)(arr);
+      //$ExpectError
+      const nthxs2: string = nth(1, ['foo'])
+      //$ExpectError
+      const nthxs3: number = nth(1, arr);
+      //$ExpectError
+      const nthxs4: number = nth(1)(arr);
+      //$ExpectError
+      const nthxs5: number = nth(1, [1])
+      //$ExpectError
+      const nthxs: ?number = nth(2)(['foo']);
+    });
+
+    it('should fail on passing a non-number to first argument', () => {
+      //$ExpectError
+      const nthxs = nth('foo')(['foo', 'bar'])
+    });
+
+    it('should fail on passing an object to second argument', () => {
```

Comment:
Minor nitpick: I'd change this from "passing an object to" to "passing a non-array to", because really we aren't specifically saying "no objects". Instead we want to assert that "we only take arrays".

Reply:
Done